File: app/src/main/java/com/example/review/New/LibraryList.java
package com.example.review.New;

import java.io.*;

public class LibraryList extends ArrayStoreList<LibraryStruct> {

    //取字节流
    @Override
    public void toBytes(DataOutputStream dos) throws IOException {
        dos.writeInt(size());
        for (LibraryStruct ls : this) {
            byte[] bytes = ls.toBytes();
            dos.writeInt(bytes.length);
            dos.write(bytes);
        }
    }

    //写字节流
    @Override
    public void loadWith(DataInputStream dis) throws IOException {
        int size = dis.readInt();
        for (int i = 0; i < size; i++) {
            int    len   = dis.readInt();
            byte[] bytes = new byte[len];
            dis.read(bytes);
            LibraryStruct libraryStruct = new LibraryStruct(bytes);
            add(libraryStruct);
        }
    }
}

File: app/src/main/java/com/example/review/New/LibraryList.kt
package com.example.review.New

import java.io.DataInputStream
import java.io.DataOutputStream
import java.io.IOException

class LibraryList : ArrayStoreList<LibraryStruct>() {
    //取字节流
    @Throws(IOException::class)
    override fun toBytes(dos: DataOutputStream) {
        dos.writeInt(size)
        for (ls in this) {
            val bytes = ls.toBytes()
            dos.writeInt(bytes.size)
            dos.write(bytes)
        }
    }

    //写字节流
    @Throws(IOException::class)
    override fun loadWith(dis: DataInputStream) {
        val size = dis.readInt()
        for (i in 0 until size) {
            val len = dis.readInt()
            val bytes = ByteArray(len)
            dis.read(bytes)
            val libraryStruct = LibraryStruct(bytes)
            add(libraryStruct)
        }
    }
}